### PR TITLE
Updates for preparing DVC/Dagshub

### PR DESCRIPTION
**Pull request by philkleer**
Changes: 

- new structure: I added new structure to the repository so that we have `logs/`, `data/`, `models/`, and `src/`
- I added `mac-requirements.txt` to include Tensorflow packages that directly support Silicon-support on M-processors (since I’m using a Mac)
- `README.md`: I updated the paths/folders and adjusted some typos
- `config.py`: I changed paths of `ROOT_DIR` (also changed from `ROOT_FOLDER` to be consistent), `DATA_DIR`, `TRAIN_PATH`, `VALID_PATH` and `TEST_PATH` to have the data now stored in the project. Created `PROC_DIR` as directory for the processed data. Added `NUM_CLASSES` automatically changed by `data_loader.py`
- `data_loader.py`\`requirements.txt`\`mac-requirements.txt` Added `shutil` (copying data from `.cache/…` to repository)and `tqdm` (progress bar for saving train/valid) as requirements 
- `data_loader.py`: Added conversion to `.TFRecord` to make it available to version the train/validation data set (functions: `save_as_tfrecord()`, `create_example()`, `_int64_feature()`, and `_bytes_feature()`). 
- `data_loader.py`: Since `num_classes` is needed with the import of train/validation data now in `train.py`, I created function `update_config_num_classes()` which will write (or update) the variable `NUM_CLASSES` in `config.py`
- `train.py`: To load the new versioned train/validation data set, I created the function `load_tfrecord_data()`. Furthermore, import from `num_classes` will now be done by the value saved in `config.py` (`NUM_CLASSES`).
- `train.py`: I added the calculation of the `steps_per_epoch` since it is not automatically captured with the new data set (`*.TFRecord`) 
- DVC initialization: I initiliazed DVC with local default (will change to Dagshub once I put the repository there)
- I added MLflow in `train.py` to save the scores for training/validation. However, I haven't had time to test it.

### CODE REVIEW

**Comment by philkleer** [comment]
So the split was a bit more complicated than I thought: I had a few issues with the label, since it was string before, which might lead to errors with tensorflow.

Furthermore, the shuffling was a bit complicated since all groups need to be represented in each subset and therefore, the shuffling should be done within the subfolders. 

I tested these new created training data sets, however, there is still somewhere a problem, since accuracy increases fast (after 90 steps is around 0.95, which is quite counterintuitive).

**Comment by philkleer** [comment]
Now the splitting works correctly! 🎉

Once you have done the dagshub connection, I can transfer files, or you run yourself the `raw_data.....py` to create the files (since right now they are just in my local space)
